tiledb/sm/filesystem/s3/STSProfileWithWebIdentityCredentialsProvider.cc

```diff
@@ -467,30 +463,40 @@ STSProfileWithWebIdentityCredentialsProvider::GetCredentialsFromWebIdentity(
     return {};
   }
 
-  Internal::STSCredentialsClient::STSAssumeRoleWithWebIdentityRequest request{
-      m_sessionName, m_roleArn, m_token};
-
-  Aws::Client::ClientConfiguration config;
-  config.scheme = Aws::Http::Scheme::HTTPS;
-  config.region = tmpRegion;
+  AssumeRoleWithWebIdentityRequest request;
+  request.SetRoleArn(m_roleArn);
+  request.SetRoleSessionName(m_sessionName);
+  request.SetWebIdentityToken(m_token);
 
-  Aws::Vector<Aws::String> retryableErrors;
-  retryableErrors.push_back("IDPCommunicationError");
-  retryableErrors.push_back("InvalidIdentityToken");
-
-  config.retryStrategy =
-      Aws::MakeShared<Aws::Client::SpecifiedRetryableErrorsRetryStrategy>(
-          CLASS_TAG, retryableErrors, 3 /*maxRetries*/);
+  auto outcome = client->AssumeRoleWithWebIdentity(request);
```

Comment:
The changes in this method's logic were to replace the use of internal APIs with public ones.

Reply:
Is there a requirement to do that now? Seeing how we've already had issues for 2 releases, in my opinion it'd be much better to minimize changes needed to solve the issue. We can then follow up with refactoring and additional testing.

If there is a high degree of confidence and there has been sufficient testing and validation of these changes, then I'm okay to move forward, but I don't like mixing defect fixes and refactoring that is not required for the defect fix.

Reply:
These are the changes that got successfully validated yesterday. We have to use the public `STSClient` class provided by the user because it has been created with the right client configuration.